src/magnify/apps/core/models.py

```diff
@@ -235,39 +235,49 @@ def jitsi_name(self):
         """The name used for the room in Jitsi."""
         return f"{self.slug:s}-{self.id!s}"
 
-    def is_administrator(self, user):
+    def get_role(self, user, direct_only=False):
         """
-        Check if a user is administrator of the room.
+        Determine the role of a given user in this room.
 
-        Users carrying the "owner" role are considered as administrators a fortiori.
+        direct_only: boolean
+            Whether or not to take into account the roles a user inherits from
+            its membership in a group.
         """
         if not user or not user.is_authenticated:
-            return False
+            return None
 
-        return (
-            self.user_accesses.filter(
-                role__in=[UserRoleChoices.ADMIN, UserRoleChoices.OWNER], user=user
-            ).exists()
-            or self.group_accesses.filter(
+        try:
+            # pylint: disable=no-member
+            role = self.user_accesses.get(user=user).role
+        except self.user_accesses.model.DoesNotExist:
+            role = None
+
+        if direct_only:
+            return role
+
+        if not (role and role != UserRoleChoices.MEMBER):
+            for access in self.group_accesses.filter(
                 Q(group__members=user) | Q(group__administrators=user),
-                role=GroupRoleChoices.ADMIN,
-            ).exists()
-        )
+            ):
+                if access.role == GroupRoleChoices.MEMBER:
+                    role = GroupRoleChoices.MEMBER
+                if access.role == GroupRoleChoices.ADMIN:
+                    role = GroupRoleChoices.ADMIN
+                    break
 
-    def is_owner(self, user):
-        """Check if a user is owner of the room."""
-        if not user or not user.is_authenticated:
-            return False
+        return role
 
-        return self.user_accesses.filter(role=UserRoleChoices.OWNER, user=user).exists()
+    def is_administrator(self, user):
+        """
+        Check if a user is administrator of the room.
 
-    def has_access(self, user):
-        """Check if a user has access to the room."""
-        return self.user_accesses.filter(
-            user=user
-        ).exists() or self.group_accesses.filter(
-            Q(group__members=user) | Q(group__administrators=user)
-        )
+        Users carrying the "owner" role are considered as administrators a fortiori.
+        """
+        return self.get_role(user) in [UserRoleChoices.ADMIN, UserRoleChoices.OWNER]
```

Comment:
```suggestion
        return self.get_role(user) in [UserRoleChoices.ADMIN, UserRoleChoices.OWNER]
```
if tomorrow GroupRoleChoices.ADMIN !== UserRoleChoices.ADMIN will this still be correct?

Reply:
Clearly, the whole concept relies on the fact that the member and administrator roles are common in group and user accesses. This is enforced in the definition of choices.

If this was modified, some tests would immediately fail so this could not be done inadvertently.

Reply:
maybe what disturbs me, is by reading the code, without further reading, I could easily think that I can only be an admin if I am an admin from my user role as it is what is written.
We could maybe just use another pair of variables that aren't attached to user or group but only target the label of admin? such as RoleChoices.ADMIN and RoleChoices.Owner and where UserRoleChoices.ADMIN will be set using RoleChoices.ADMIN and the same for GroupRoleChoices
wdyt?

Reply:
Same. The refactoring in https://github.com/openfun/jitsi-magnify/pull/135/commits/b8d2a013555a7b1a4fcbada65737d7c7b1f5380a should please you.